.github/workflows/build-image.yaml

```diff
@@ -0,0 +1,74 @@
+name: Build Image
+on:
+  pull_request:
+    types: [opened, synchronize, reopened, labeled]
+  push:
+    branches:
+      - main
+      - release/**
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+jobs:
+  build-image:
+    name: Build and publish uptime-checker to GHCR
+    if: ${{ !github.event.pull_request.head.repo.fork && github.actor != 'dependabot[bot]' }}
+    runs-on: ubuntu-latest
+
+    env:
+      VERSIONED_IMAGE: ghcr.io/getsentry/uptime-checker:${{ github.sha }}
+      NIGHTLY_IMAGE: ghcr.io/getsentry/uptime-checker:nightly
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: docker/setup-qemu-action@v3
+      - uses: docker/setup-buildx-action@v3
+
+      - name: Login to GHCR
+        run: docker login --username '${{ github.actor }}' --password '${{ secrets.GITHUB_TOKEN }}' ghcr.io
+
+      - name: Build and push to GHCR
+        run: >-
+          docker buildx build
```

Comment:
I really do not like us double building this image (once in Google Cloud and once in GHA). Looks like a big waste of resources. @asottile-sentry do you know if we have any plans to address this?